demo/src/main/java/com/novoda/merlin/DemoActivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.novoda.merlin
+
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.view.View
+import android.widget.Toast
+import kotlinx.android.synthetic.main.activity_demo.*
+
+class DemoActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_demo)
+
+        current_status.setOnClickListener(reportNetworkStatus())
+        has_internet_access.setOnClickListener(reportInternetAccess())
+        wifi_connected.setOnClickListener(reportWifiConnected())
+        mobile_connected.setOnClickListener(reportMobileConnected())
+        network_subtype.setOnClickListener(reportNetworkSubtype())
+        next_activity.setOnClickListener(navigateToNextActivity())
+    }
+
+    private fun reportNetworkStatus(): View.OnClickListener {
```

Comment:
These will probably be collapsed into a single function call later where we supply a message and whether it was positive or negative. I'll do this when I introduce the themer for the snackbar again.